src/nodes/gradient.ts

```diff
@@ -25,14 +28,38 @@ export abstract class Gradient extends NonRenderedNode {
       return
     }
 
+    // Only need to calculate the ancestors array once
+    if (!this.ancestors) {
+      this.ancestors = []
+      let ancestor: SvgNode | null = this as SvgNode
+      while (ancestor) {
+        this.ancestors.push(ancestor)
+        ancestor = ancestor.getParent()
+      }
+      // Ensure the top level ancestor comes first in the array
+      this.ancestors = this.ancestors.reverse()
+    }
+
+    let contextColor: RGBColor | null = null
+    this.ancestors.forEach(a => {
+      const colorAttr = getAttribute(a.element, context.styleSheets, 'color')
+      if (colorAttr) {
+        contextColor = parseColor(colorAttr, null)
+      }
+    })
+
```

Comment:
Instead of the ancestors array, we can store the contextColor as field, right? Then we can condense this into a single loop and stop as soon as we find a 'color'  attribute.

Reply:
Yep, you're quite correct. Silly me thinking in a top-down way when bottom-up would make more sense here. I can just traverse upwards once till I find the 'closest' `color` attribute.
I'll push a commit ASAP.

Reply:
Pushed a commit that replaces the `ancestors` member with `contextColor`. Starts off as `undefined` but the first time `apply` is called, the ancestors are traversed upwards to calculate the 'closest' color. Any subsequent calls just use this same value without traversing again.